habitat_baselines/rl/ddppo/ddp_utils.py

```diff
@@ -125,57 +145,52 @@ def add_signal_handlers() -> None:
     # the job ample time to cleanup and exit.
     signal.signal(signal.SIGUSR2, _clean_exit_handler)
 
+    # SLURM always sends SIGTERM so we can use this to save and exit
+    signal.signal(signal.SIGTERM, _clean_exit_and_save_handler)
+
     signal.signal(signal.SIGUSR1, _requeue_handler)
 
 
 @rank0_only
-def save_interrupted_state(state: Any, filename: str = None):
-    r"""Saves the interrupted job state to the specified filename.
+def save_resume_state(state: Any, filename_or_config: Union[Config, str]):
+    r"""Saves the resume job state to the specified filename.
         This is useful when working with preemptable job partitions.
 
-    This method will do nothing if SLURM is not currently being used and the filename is the default
-
     :param state: The state to save
-    :param filename: The filename.  Defaults to "${HOME}/.interrupted_states/${SLURM_JOBID}.pth"
+    :param filename_or_config: The filename of the saved state or the config to construct it.
     """
-    if SLURM_JOBID is None and filename is None:
-        logger.warn("SLURM_JOBID is none, not saving interrupted state")
-        return
-
-    if filename is None:
-        filename = INTERRUPTED_STATE_FILE
-        if not osp.exists(osp.dirname(INTERRUPTED_STATE_FILE)):
-            raise RuntimeError(
-                "Please create a .interrupted_states directory in your home directory for job preemption"
-                "(This is intentionally not created automatically as it can get quite large)"
-            )
+    if isinstance(filename_or_config, Config):
+        filename = resume_state_filename(filename_or_config)
+    else:
+        filename = cast(str, filename_or_config)
```

Comment:
Instead of casting here, can't you just do ```filename = str(filename_or_config)``` or do assert isinstance(filename_or_config, str)

Reply:
The cast isn't needed here, mypy already knows that filename_or_config is an str, I just added the cast because it makes it a little clearer, but I'll remove it.  I don't think actually forcing this to be a string is the right call because this is python!

Reply:
@erikwijmans You can do that with a comment or a filename : str = ... . 

cast overrides the typing inference which could lead to it eating typing errors later one.